mne/chpi.py

```diff
@@ -1615,3 +1625,310 @@ def get_active_chpi(raw, *, on_missing="raise", verbose=None):
+
+    The algorithm is as follows:
+
+    1. Optionally recompute HPI amplitudes (sinusoidal fit for each channel) using
+       :func:`~mne.chpi.compute_chpi_amplitudes`.
+    2. Optionally use HPI amplitudes to fit HPI coil locations using
+       :func:`~mne.chpi.compute_chpi_locs`.
+    3. Optionally determine coil digitization order by testing all permutations
+       for the best goodness of fit between digitized coil locations and
+       (rigid-transformed) fitted coil locations.
+    4. Subselect coils to use for fitting ``dev_head_t`` based on ``gof_limit``,
+       ``dist_limit``, and ``max_use``.
+    5. Update info inplace by modifying ``info["dev_head_t"]`` and appending new entries
+       to ``info["hpi_meas"]`` and ``info["hpi_results"]``.
+
+    .. versionadded:: 1.11
+    """
+    _validate_type(info, Info, "info")
+    fit_info = pick_info(info, pick_types(info, meg=True, exclude=()))
+    # Set bads to empty list here. In theory flux jumps etc. or even flat channels
+    # shouldn't affect the fit much. At some point we could allow ignoring bads,
+    # but it would make the API more complex (KISS) and make the info accounting harder
+    # (e.g., slopes must always have shape[-1] == len(all_meg_chs)).
+    fit_info["bads"] = []  # for backward compat... maybe shouldn't do this
+    vf = _verbose_safe_false()
+    hpi = _setup_hpi_amplitude_fitting(fit_info, 1.0, ext_order=ext_order, verbose=vf)
+    n_coils = info["hpi_subsystem"]["ncoil"]
+    if use is not None:
+        use = np.array(use)
+        if use.dtype.kind != "i":
+            raise ValueError(
+                f"use must be an integer or array-like of integers, got {use.dtype}"
+            )
+        if use.shape == ():
+            use = int(use.item())
+            if use < 3:
+                raise ValueError(f"max_use must be at least 3, got {use}")
+        elif (
+            use.ndim != 1
+            or not np.array_equal(np.sort(use), np.unique(use))
+            or not np.isin(use, np.arange(info["hpi_subsystem"]["ncoil"])).all()
+        ):
+            raise ValueError(
+                "use must be a 1D array of unique integers in the range [0, "
+                f"{n_coils - 1}]"
+            )
+    assert use is None or isinstance(use, int | np.ndarray)  # we have this now
+    _validate_type(amplitudes, bool, "amplitudes")
+    _validate_type(locs, bool, "locs")
+    _validate_type(order, bool, "order")
+    _validate_type(dist_limit, "numeric", "dist_limit")
+    _validate_type(linearity_limit, "numeric", "linearity_limit")
+    if amplitudes and not locs:
+        raise ValueError(
+            "If amplitudes is True, locs must also be True (otherwise "
+            "recomputing amplitudes has no effect)"
+        )
+    if locs and not order:
+        raise ValueError(
+            "If locs is True, order must also be True (otherwise "
+            "recomputing locations has no effect)"
+        )
```

Comment:
I'm +0.5 for adding this info to the docstring as well, just to save diligent readers one cycle of run-error-debug-rerun